Pass 2 of addressing review comments in ITE-2

adityasaky opened the pull request:
I still need to structure the root-layouts-can-differ in, I'll push here shortly.

adityasaky committed:
(WIP) Pass 2 of addressing review comments

Signed-off-by: Aditya Sirish <aditya@saky.in>

adityasaky committed:
(WIP) Mention root layout MAY be identical

Signed-off-by: Aditya Sirish <aditya@saky.in>

adityasaky commented on ITE/2/README.adoc:
```diff
@@ -169,21 +169,23 @@ are three important considerations regarding the targets metadata signed
-different in-toto link metadata for different packages, where the name
-MAY be the SHA-256 hash of the "`developer`" step link metadata file for
-a particular package.
+. Since each package for a project is typically produced independently
```
I added "for a project" to help clarify that different packages are still from the same abstract supply chain.

adityasaky commented on ITE/2/README.adoc:
```diff
@@ -169,21 +169,23 @@ are three important considerations regarding the targets metadata signed
+ways, but perhaps the simplest is to use directories with unique names to
+isolate different in-toto link metadata for different packages, where the
+name MAY be the SHA-256 hash of the "`developer`" step link metadata file
+for a particular package. The root layout MAY be identical for each
```
Does this bit make sense?

trishankatdatadog commented on ITE/2/README.adoc:
```diff
@@ -169,21 +169,23 @@ are three important considerations regarding the targets metadata signed
+isolate different in-toto link metadata for different packages, where the
+name MAY be the SHA-256 hash of the "`developer`" step link metadata file
+for a particular package. The root layout MAY be identical for each
+of these packages, but it can differ to allow for changes in the supply
```
Could we please mention here that using consistent snapshots and custom targets metadata in TUF lets us trivially associate different packages (whether in the same project or not) with different root layout w/o breaking backwards compatibility?

adityasaky replied:
Does https://github.com/trishankatdatadog/ITE/pull/3/commits/df8c94889d1605bd49697d266db18a83a161abee look ok?

adityasaky replied:
Actually not so much I think. Let me try again.

trishankatdatadog reviewed (commented):
This is excellent, thanks, Aditya! Would you please kindly address my comment?

adityasaky committed:
(WIP) Mention possibility of entirely different root layouts

This is for situations where different packages are results of
different projects / supply chains.

Signed-off-by: Aditya Sirish <aditya@saky.in>

trishankatdatadog reviewed (approved):
💯, thanks, Aditya!
